### ISSUE TITLE
Projects module depends on themes and Actions

### ISSUE THREAD

**Issue by tzargoth** [gabdub/ta-tweaks#8]
Documentation states to only require util and projects in init.lua.  Doing so results in a multitude of errors.  Using other themes provided for TA results in similar/same errors.  I'd really like to use an alternate dark theme without having to hack the project's dependent theme files ... this does not appear to be trivial.

Also needs "plugs = {}" in the init.lua file.

If I comment out the 3 lines in proj_ui.lua requiring the theme and include the "textadept" module, I see the Projects menu item, but all three frames are blank buffers, and Projects hasn't loaded properly.  Attempting Project > New results in "Unknown command: new_project".

I am using TextAdept 10.8 with latest source from here, on Linux.

**Comment by gabdub** [comment]
You are right. I made many changes since the initial project code was released.
Now it **requires actions** to show the Project menu. I'll update the wiki when I have some time.

This are the files I copy in ~/.textadept to get something running with the **dark** theme:
**lexers**, **themes**, **modules/project**, **modules/util**, **modules/textadept**
I put this in ~/.textadept/**init.lua**:
```
plugs= {}
local p= buffer.property
p['color.curr_line_back']=0xf5f9ff
p['color.prj_sel_bar_nof']=0xe0e0e0
p['color.prj_sel_bar']=0xffffff
p['color.prj_open_mark']=0x404040
require('util')
for _, buff in ipairs(_BUFFERS) do buff:set_theme('dark') end
require('project')
```
I also commented lines 717 and 720 in proj_ui.lua (the function Proj.set_contextm_search() is not defined).

I'm afraid this is all I can do right now.

Cheers,
Gabriel

**Comment by tzargoth** [comment]
Thanks ... this gets dark theme going.  However, I still only see 3 empty frame buffers instead of projects info.  There are no errors unless I attempt to use the project functions from the menu.  I have commented out contextm function calls on lines 716 and 719 ... is the line offset somehow an issue?  If you are seeing and able to use projects functions, I wonder what the difference is.  I'll try again, replacing all my files from originals ... maybe I messed something up.

No change ... still get the same empty buffers.

I added "require('textadept')" to my init.lua and now it doesn't give errors for Project > New ... but frames are still empty when first loaded.  Not sure if project functions work ... will test.

I uncommented those lines in proj_ui.lua since those function definitions are dependent on actions, which only gets loaded when textadept module is loaded.  I used the Project > New function and it seems to be working.  If the frame buffers are supposed to be empty on first load, then I guess things are working as intended.  The colors need some tweaking, since files and folders in the project list are black on dark background.

Figured out the theme colors from myproj.lua lexer ... ended up with these colors in my init.lua:
```
    p['color.curr_line_back']=0x444444
    p['color.prj_sel_bar_nof']=0x444444
    p['color.prj_sel_bar']=0x444444
    p['color.prj_open_mark']=0x222222
    -- file extensions
    p['color.comment']=0xFFAAAA
    -- folders
    p['color.function']=0xAAFFFF
    -- project name
    p['color.variable']=0xAAAAFF
```
I am assuming 'color.type' is not used, or not needed for what project info is shown.
